nf_core/components/install.py

```diff
@@ -47,6 +50,12 @@ def __init__(
             self.installed_by = [self.component_type]
 
     def install(self, component: str, silent: bool = False) -> bool:
+        if isinstance(component, dict):
```

Comment:
If the `isinstance` check is needed (and indeed even if it isn't - because this is always a dict now) then the type hint in the method signature is incorrect.

Reply:
From what I could see, this is the single `isinstance` check that was actually needed. There are some places in the codebase where this function gets called with a string input, when removing the isinstance check I noticed quite a few test failures. So, to aim for minimal impact and changes to the codebase and current implementations, I've opted to change the argument type.